static/js/apps/eval_rig/feedback_form.tsx

```diff
@@ -14,181 +14,165 @@
-          <h3>Response from Data Commons</h3>
-          <label>
-            <input
-              type="radio"
-              name="dcResponse"
-              value="0"
-              checked={responses.dcResponse === "0"}
-              onChange={handleChange}
-            />
-            Relevant and direct
-          </label>
-          <label>
-            <input
-              type="radio"
-              name="dcResponse"
-              value="1"
-              checked={responses.dcResponse === "1"}
-              onChange={handleChange}
-            />
-            Doesn&apos;t match the question
-          </label>
+          <OneQuestion
+            question="Response from Data Commons"
+            name="dcResponse"
+            values={{
+              DC_ANSWER_RELEVANT: "Relevant and direct",
+              DC_ANSWER_IRRELEVANT: "Doesn't match the question",
+              DC_ANSWER_EMPTY_BADNL: "Data exists, but NL fails to respond",
+              DC_ANSWER_EMPTY_NODATA:
+                "Query asks for data that doesn't exist in DC",
+              DC_ANSWER_EMPTY_OUTOFSCOPE:
+                "Query asks for data that is out-of-scope for DC",
+            }}
+            handleChange={handleChange}
+            responseField={response.dcResponse}
+          />
 
-          <h3>Data Commons stat quality</h3>
-          <label>
-            <input
-              type="radio"
-              name="dcStat"
-              value="0"
-              checked={responses.dcStat === "0"}
-              onChange={handleChange}
-            />
-            Stats Seem Accurate
-          </label>
-          <label>
-            <input
-              type="radio"
-              name="dcStat"
-              value="1"
-              checked={responses.dcStat === "1"}
-              onChange={handleChange}
-            />
-            Stat values seem off
-          </label>
+          <OneQuestion
+            question="Response from Data Commons"
+            name="dcStat"
+            values={{
+              DC_STAT_ACCURATE: "Stats seem accurate",
+              DC_STAT_INACCURATE: "Stats seem accurate",
+            }}
+            handleChange={handleChange}
+            responseField={response.dcStat}
+          />
         </div>
 
-        <button type="submit">Submit</button>
+        <button type="submit" disabled={Object.values(response).includes("")}>
+          Submit
+        </button>
       </fieldset>
     </form>
   );
 }
 
 // Mock function to simulate API call to save responses
-function saveResponses(responses) {
-  console.log("API Call to save responses:", responses);
+async function saveResponse(
+  sheetId: string,
+  queryId: string,
+  callId: string,
+  response: Response
+) {
+  try {
+    // Define the document reference
+    const docRef = doc(
+      db,
+      "sheets",
+      sheetId,
+      "queries",
+      queryId,
+      "calls",
+      callId
+    );
+    // Save the data to Firestore
```

Comment:
should we also have userId column in the key?

Reply:
since one query/call be evaluated by different users. How about making the response a collection as well, so it can contain multiple evaluations from same/different users?

Reply:
That makes sense to me!  Though I don't understand what you mean by collection, in this context.

Reply:
Oh they are firebase data model, basically to form the hierarchy data structure: https://firebase.google.com/docs/firestore/data-model